### PR TITLE
Add tests for CustomDropdown widget (CAT-1566)

### PR DESCRIPTION
**Pull request by dtitenko**

### CODE REVIEW

File: src/components/CustomDropdownComp.tsx [hunk 1]
```diff
@@ -33,161 +33,157 @@ export interface CustomDropdownComponentProps {
...
+                }
+                if (props.createValue.canExecute) {
+                    props.createValue.execute();
+                }
+                if (!props.enableCreate) {
+                    setValue(createOption(inputValue.value, '', ''));
+                    setOptions([...options, createOption(inputValue.value, '', '')])
+                }
+            } catch (err) {
+                console.error('Failed to create a Tag: ' + err);
```
Comment (reinouts) [inline src/components/CustomDropdownComp.tsx:114]: Is there no more graceful way to handle this case than to log an error on the console?

File: src/components/CustomDropdownComp.tsx [hunk 2]
```diff
@@ -33,161 +33,157 @@ export interface CustomDropdownComponentProps {
...
+                if (props.contextObjLabel.status === 'available') {
+                    props.contextObjLabel.setValue(actionMeta.removedValues[0].value);
+                }
+                if (props.clearValue.canExecute) {
+                    props.clearValue.execute();
+                }
+                setValue(null);
             } catch (err) {
-              console.error('Failed to create a Tag: ' + err);
+                console.error('Failed to create a Tag: ' + err);
```
Comment (reinouts) [inline src/components/CustomDropdownComp.tsx:129]: Same question about graceful error handling

File: src/components/__tests__/CustomDropdown.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        { value: { displayValue: 'item3' } },
+    ]
+};
+
+const renderComponent = (override = {}) => {
+    const props = {
+        defaultValue: createOption(),
+        firstLabel: (item) => item.value,
+        secondLabel: (item) => item.value,
+        imgUrl: (item) => item.value,
```
Comment (reinouts) [inline src/components/__tests__/CustomDropdown.spec.tsx:28]: What is `item.value` in these three cases, that you initialize firstLabel, secondLabel and imgUrl to the same thing?

File: src/components/CustomDropdownComp.tsx [hunk 4]
```diff
@@ -33,161 +33,157 @@ export interface CustomDropdownComponentProps {
...
-                EnableAvatar={props.useAvatar}
-                SecondLabel={secondaryLabel}
-                classNameCreateModal={"xxx"}
-            />,
+            DisplayName={label}
+            UrlString={urlstring}
+            ClassNamePrefix={props.classNamePrefix}
+            EnableAvatar={props.useAvatar}
+            SecondLabel={secondaryLabel}
+            classNameCreateModal={"xxx"}
```
Comment (reinouts) [inline src/components/CustomDropdownComp.tsx:51]: Is the `xxx` supposed to be there?

File: src/components/__tests__/CustomDropdown.spec.tsx (hunk 3 above)
Comment (dtitenko) [reply src/components/__tests__/CustomDropdown.spec.tsx:28]: It's the very beginning of the mock around the widget. All three will take the same and only value from the option provided to the widget. Of course the accessors has to be different to test whether accessors are being used.

File: src/components/__tests__/CustomDropdown.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    })
+
+    describe('and clear button is pressed', () => {
+        let component;
+        const clearValue = jest.fn();
+        const contextObjLabelSetValue = jest.fn();
+        const contextObjIdSetValue = jest.fn();
+
+        beforeEach(() => {
+            const comp = renderComponent({
```
Comment (ysmendix) [inline src/components/__tests__/CustomDropdown.spec.tsx:113]: can use component, no need for creating a new variable

File: src/components/CustomDropdownComp.tsx [hunk 6]
```diff
@@ -4,7 +4,7 @@ import CreatableSelect from 'react-select/creatable';
 import { Styles } from 'react-select/src/styles';
 import { OptionTypeBase } from "react-select/src/types";
 
-import { ListValue, EditableValue, ActionValue, ListAttributeValue } from "mendix";
+import { ListValue, EditableValue, ActionValue, ListAttributeValue, ValueStatus } from "mendix";
```
Comment (ysmendix) [inline src/components/CustomDropdownComp.tsx:7]: ValueStatus is not used

File: src/components/CustomDropdownComp.tsx [hunk 7]
```diff
@@ -35,7 +35,7 @@ export interface CustomDropdownComponentProps {
     className: string;
     classNamePrefix: string;
     menuHeight: number;
-    onFocus: ActionValue;
+    onFocus?: ActionValue;
```
Comment (ysmendix) [inline src/components/CustomDropdownComp.tsx:38]: Why do we need changes in this file? Do they relate to this story?

Comment (dtitenko) [reply src/components/CustomDropdownComp.tsx:38]: I saw from the code that it is optional, so has to be declared as such. From client (developer) perspective, she might not want to pass this handler and not interested in handling this.
```
const handleFocus = () => {
        if (props.onFocus != undefined && props.onFocus.canExecute) {
            props.onFocus.execute();
        }
    }
```

**Review by ysmendix — APPROVED** [review]

**Pull request merged**